sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/sqlite3_partition_manager.py

```diff
@@ -0,0 +1,117 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+import time
+import uuid
+import sqlite3
+from .partition_manager import PartitionManager
+
+
+class Sqlite3PartitionManager(PartitionManager):
+    """An implementation of PartitionManager by using the sqlite3 in Python standard library.
+    Sqlite3 is a mini sql database that runs in memory or files.
+
+
+    """
+    def __init__(self, db_filename, ownership_table="ownership"):
+        """
+
+        :param db_filename: name of file that saves the sql data.
+        Sqlite3 will run in memory without a file when db_filename is ":memory:".
+        :param ownership_table: The table name of the sqlite3 database.
+        """
+        super(Sqlite3PartitionManager, self).__init__()
+        self.ownership_table = ownership_table
+        conn = sqlite3.connect(db_filename)
+        c = conn.cursor()
+        try:
+            c.execute("create table " + ownership_table +
```

Comment:
This code is very likely vulnerable to SQL injection attacks.

Reply:
Only table name is concatinated. I can't find a way to parameterize table name instead of concatination for now.
We can write a function to remove all special characters from "ownership_table" as a defense of sql injection.  ## internal stuff, not urgent for now 
For the query parameters, I've already parameterized them.

Reply:
I would rather be overly restrictive in the names allowed (e.g. only a-zA-Z, no spaces) than publish with an injection vector.

Reply:
restricted table name to characters that can't cause sql injection